src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -0,0 +1,361 @@
+        p_medication = self.calculate_medication_coverage_probabilities(pop)
+        medicated_states = self.randomness.choice(
+            p_medication.index, choices=p_medication.columns, p=np.array(p_medication)
+        )
+        medicated_sbp = medicated_states[medicated_states.isin(["sbp", "both"])].index
+        medicated_ldlc = medicated_states[medicated_states.isin(["ldlc", "both"])].index
+        # Define what level of medication for the medicated simulants
+        pop.loc[medicated_sbp, self.sbp_medication_column] = self.randomness.choice(
+            medicated_sbp,
+            choices=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["sbp"].keys()),
+            p=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["sbp"].values()),
+        )
+        pop.loc[medicated_ldlc, self.ldlc_medication_column] = self.randomness.choice(
+            medicated_ldlc,
+            choices=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["ldlc"].keys()),
+            p=list(data_values.BASELINE_MEDICATION_LEVEL_PROBABILITY["ldlc"].values()),
+        )
+
+        # # Move medicated but non-adherent simulants to lowest level
+        sbp_non_adherent = pop[
+            pop[self.sbp_medication_adherence_type_column]
+            != data_values.MEDICATION_ADHERENCE_TYPE.ADHERENT
+        ].index
+        ldlc_non_adherent = pop[
+            pop[self.ldlc_medication_adherence_type_column]
+            != data_values.MEDICATION_ADHERENCE_TYPE.ADHERENT
+        ].index
+        pop.loc[medicated_sbp.intersection(sbp_non_adherent), self.sbp_medication_column] = 1
+        pop.loc[
+            medicated_ldlc.intersection(ldlc_non_adherent), self.ldlc_medication_column
+        ] = 1
+
+        return pop
+
+    def calculate_medication_coverage_probabilities(self, pop: pd.DataFrame) -> pd.DataFrame:
+        """Determine the probability of each simulant being medicated"""
+
+        # Calculate the covariates
+        medication_coverage_covariates = {}
+        for (
+            name,
+            c_int,
+            c_sbp,
+            c_ldlc,
+            c_age,
+            c_sex,
+        ) in data_values.MEDICATION_COVERAGE_COEFFICIENTS:
+            medication_coverage_covariates[name] = np.exp(
+                c_int
+                + c_sbp * self.sbp(pop.index)
+                + c_ldlc * self.ldlc(pop.index)
+                + c_age * pop["age"]
+                + c_sex * pop["sex"].map(data_values.BASELINE_MEDICATION_COVERAGE_SEX_MAPPING)
+            )
+        # Calculate probabilities of being medicated
+        p_medication = {}
+        p_denominator = sum(medication_coverage_covariates.values()) + 1
+        for med_type, cov in medication_coverage_covariates.items():
+            p_medication[med_type] = pd.Series(cov / p_denominator, name=med_type)
+        p_medication["none"] = pd.Series(1 / p_denominator, name="none")
+
+        return pd.concat(p_medication, axis=1)
```

Comment:
Im not sure I completely understand the math that is happening here but the passing of objects along looks fine with no errors.